lambda/src/index.js

```diff
@@ -15,20 +15,23 @@ exports.handler = async function (event, context) {
   // Give the ALB 10 seconds to make sure the test TG has switched to the new code.
   const timer = sleep(10000)
 
+  let errorFromDownloads
   if (process.env.POSTMAN_API_KEY) {
     // download postman files from Postman API
     await Promise.all([
       downloadFileFromPostman('collection', process.env.POSTMAN_COLLECTION_NAME),
       downloadFileFromPostman('environment', process.env.POSTMAN_ENVIRONMENT_NAME)
-    ])
+    ]).catch(err => { errorFromDownloads = err })
   } else {
     // download postman files from S3 Bucket
     await Promise.all([
       downloadFileFromBucket('collection', process.env.POSTMAN_COLLECTION),
       downloadFileFromBucket('environment', process.env.POSTMAN_ENVIRONMENT)
-    ])
+    ]).catch(err => { errorFromDownloads = err })
   }
 
+  if (errorFromDownloads) throw errorFromDownloads // cause lambda to fail if collection/env failed to download to lambda
```

Comment:
These changes here are unnecessary - the error (a rejected promise) will just bubble up on its own.

We did that kind of pattern down below because we wanted to run CodeDeploy even if the tests failed.

Reply:
The error then says something like "uncaught error in the promise," I thought I had to catch them. But you're right, I do want codedeploy portion to run if these fail.

Reply:
I had to keep track of the errors or it wouldn't even run the codedeploy section. Don't we want CodeDeploy to fail if the collection wasn't found?

Reply:
>The error then says something like `UnhandledPromiseRejectionWarning`

Yeah. That's what it would say if you were running the code locally, but IIRC, Lambda's invoker "catches" those. ["When you resolve or reject the promise, Lambda sends the response or error to the invoker."](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html)

Reply:
And yeah, we _do_ want to tell CodeDeploy that we failed if the downloads failed. Good catch with that.